toolchain/check/import_cpp.cpp

```diff
@@ -967,6 +979,9 @@ static auto ImportFunctionDecl(Context& context, SemIR::LocId loc_id,
     MarkFailedDecl(context, clang_decl);
     return SemIR::ErrorInst::InstId;
   }
+  // TODO: Convert to context.TODO() when we have a test case.
+  CARBON_CHECK(!clang_decl->getFunctionType()->isFunctionNoProtoType(),
+               "No Prototype function");
```

Comment:
Seems fine to remove the `TODO` here. This can only happen in non-C++ compiles (C, Objective-C, and similar); C++ code only ever uses `FunctionProtoType`. Just CHECK-failing on this seems fine given that's what we do for (eg) `RecordDecl`s that aren't `CXXRecordDecl`s (which also only happens in non-C++ compiles).

It might also make sense to reverse the sense of this and check `isFunctionProtoType` -- we need it to be a `FunctionProtoType`, and if it's not we don't care whether it's a `FunctionNoProtoType` or some other kind of type.

Reply:
Done.